Python Operator plagiarism (#499)
Codecademy/docs: content/python/concepts/operators/operators.md

  - 'paths/data-science'
---

> Operators are used to perform various operations on variables and values.
> The standard arithmetic and assignment operators are the most familiar.

Reviewer: ```suggestion
Operators are used to perform various operations on variables and values. The standard arithmetic and assignment operators are the most familiar.
```

> The following bit of Python code uses a multiplication operator, `*`, to calculate the product of two numbers,
> and the assignment operator, `=` to store the result in the variable `x`.

Reviewer: ```suggestion
## Syntax

The following code snippet uses an arithmetic operator for multiplication, `*`, to calculate the product of two numbers, along with the assignment operator, `=`, to store the result in the variable `x`.
```

```py
# Store five times three in variable x
x = 5 * 3
```

Python operators can be organized into the following groups:

> - Arithmetic operators

Reviewer: ```suggestion
- Arithmetic operators for performing traditional math evaluations.
```

> - Assignment operators

Reviewer: ```suggestion
- Assignment operators for assigning values to variables.
```

  - 'paths/data-science'
---

Operators are used to perform various operations on variables and values. The standard arithmetic and assignment operators are the most familiar.

## Syntax

The following code snippet uses an arithmetic operator for multiplication, `*`, to calculate the product of two numbers, along with the assignment operator, `=`, to store the result in the variable `x`.

```py
# Store five times three in variable x

> x = 5 * 3

Reviewer 2: I have an after-thought which is to make this a bit more reusable.

> The following code snippet assigns `my_variable` to the value of `num1` and `num2` with some arithmetic operator acting on them. For example, if the operator represented an asterisk, `*`,  the `my_variable` would be assigned a value of `num1 * num2`.

```
my_variable = num1 operator num2... and so on
```

Reviewer 2: Maybe we can mention order of operations as well

Author: @yangc95 did both.

- Equal, `==`, for returning `True` if two values are equal.
- Not equal, `!=`, for returning `True` if two values are not equal.
- Less than, `<`, for returning `True` if left value less than right value.
- Less than or equal to, `<=`, for returning `True` if left value is less than or equal to right value.
- Greater than, `>`, for returning `True` if left value greater than right value.
- Greater than or equal to, `>=`, for returning `True` if left value greater than or equal to right value.

## Logical Operators

Python has the following logical operators:

- The `and` operator returns `True` if both statements are `True`.
- The `or` operator returns `True` if either statement is `True`.
- The `not` operator returns `True` if its associated statement is `False`.

## Order of Operations

Python evaluates an expression in order of precedence as follows:

- Items in parentheses, (`(`...`)`), have the highest level of precedence, expressions within them are evaluated first.
- Exponentiation (`**`)
- Multiplication and division operators (`*`, `/`, `//` & `%`)
- Addition and subtraction (`+` & `-`)
- Comparison (`<`, `<=`, `>` & `>=`)
- Equality (`==` & `!=`)
- `not`
- `and`
- `or`

> - Items at the same precedence are evaluated left to right. The exception to this is exponentiation, which evaluates right to left.

Reviewer 2: ```suggestion

**Note:** Items at the same precedence are evaluated left to right. The exception to this is exponentiation, which evaluates right to left.
```

## Arithmetic Operators

Python has the following arithmetic operators:

> - `+`: Addition, returns the sum of two numbers.
> - `-`: Subtraction, returns the difference of two numbers.
> - `*`: Multiplication, returns the product of two numbers.
> - `/`: Division, returns the quotient of two numbers.
> - `**`: Exponentiation, returns the value of one number raised to the power of another.
> - `%`: Modulus, returns the remainder of one number divided by another.
> - `//`: Floor division, returns integer quotient of two numbers.

Reviewer: ```suggestion
- Addition, `+`, for returning the sum of two numbers.
- Subtraction, `-`, for returning the difference of two numbers.
- Multiplication, `*`, for returning the product of two numbers.
- Division, `/`, for returning the quotient of two numbers.
- Exponents, `**`, for returning the value of one number raised to the power of another.
- Modulus, `%`, for returning the remainder of one number divided by another.
- Floor division, `//`, for returning the integer quotient of two numbers.
```

Reviewer: What do you think sounds/reads better? "for returning" or "returns/which returns"?

Reviewer 2: My opinion is to actually keep the symbol operators before the text because it reads clearer.

Also, I think "which returns" sounds better. "`+`: Addition symbol, which returns..."

## Assignment Operators

Python includes the following assignment operators:

> - `=`: Assigns the value on the left to the variable on the right.
> - `+=`: Adds the value on the right to the variable on the left then assigns the result to the variable on the left.
> - `-=`: Subtracts the value on the right from the variable on the left then assigns the result to the variable on the left.
> - `*=`: Multiplies the value on the right with the variable on the left then assigns the result to the variable on the left.
> - `/=`: Divides the value on the right into the variable on the left then assigns the result to the variable on the left.
> - `%=`: Calculates the modulus of the variable on the right with the value on the left and assigns the result to the variable on the left.

Reviewer: ```suggestion
- The `=` operator assigns the value on the left to the variable on the right.
- The `+=` operator updates a variable by incrementing its value and reassigning it.
- The `-=` operator updates a variable by decrementing its value and reassigning it.
- The `*=` operator updates a variable by multiplying its value and reassigning it.
- The `/=` operator updates a variable by dividing its value and reassigning it.
- The `%=` operator updates a variable by calculating its modulus against another value and reassigning.
```

Reviewer 2: Hmm, I guess this can read more fluidly for some. I'm not opposed to making these (and the above list) full sentences.

Reviewer: Totes fine with either. Your call. 🙂

## Comparison Operators

Python has the following comparison operators:

> - `==`: Equal, `True` if two values are equal.
> - `!=`: Not equal, `True` if two values are not equal.
> - `<`: Less than, `True` if left value less than right value.
> - `<=`: Less than or equal to, `True` if left value is less than or equal to right value.
> - `>`: Greater than, `True` if left value greater than right value.
> - `>=`: Greater than or equal to, `True` if left value greater than or equal to right value.

Reviewer: ```suggestion
- Equal, `==`, for returning `True` if two values are equal.
- Not equal, `!=`, for returning `True` if two values are not equal.
- Less than, `<`, for returning `True` if left value less than right value.
- Less than or equal to, `<=`, for returning `True` if left value is less than or equal to right value.
- Greater than, `>`, for returning `True` if left value greater than right value.
- Greater than or equal to, `>=`, `True` if left value greater than or equal to right value.
```

## Logical Operators

Python has the following logical operators:

> - `and` : Returns `True` if both statements are `True`.
> - `or` : Returns `True` if either statement is `True`.
> - `not` : Returns `True` if following statement is `False`.

Reviewer: ```suggestion
- The `and` operator returns `True` if both statements are `True`.
- The `or` operator returns `True` if either statement is `True`.
- The `not` operator returns `True` if its associated statement is `False`.
```
